src/es/customer-loyality/organisms/form/Form.js

```diff
@@ -0,0 +1,213 @@
+/* global CustomEvent */
+import { Shadow } from '../../../components/web-components-toolbox/src/es/components/prototypes/Shadow.js'
+import { escapeForHtml } from '../../helpers/Shared.js'
+
+/**
+ * Voting Option
+ *
+ * @export
+ * @class Option
+ * @type {CustomElementConstructor}
+ */
+
+export default class Form extends Shadow() {
+  /**
+   * @param {any} args
+   */
+  constructor (options = {}, ...args) {
+    super({ importMetaUrl: import.meta.url, ...options }, ...args)
+  }
+
+  connectedCallback () {
+    const voting = this.dataset.voting ? JSON.parse(this.dataset.voting) : null
+    if (voting) {
+      if (this.shouldRenderCSS()) this.renderCSS()
+      this.renderHTML(voting)
+    }
+
+    this.submitListener = (evt) => {
+      evt.preventDefault()
+      const formData = new FormData(this.form)
+      const params = new URLSearchParams(window.location.search)
+      this.form.dispatchEvent(new CustomEvent('submit-voting', {
+        bubbles: true,
+        cancelable: true,
+        composed: true,
+        detail: {
+          kursId: voting.course.id,
+          teilnehmerId: params.get('teilnehmerId'),
+          optionPriceAvailable: voting.optionPrice.available,
+          optionPriceValue: formData.get('optionPrice') === 'on',
+          optionLessonsAvailable: voting.optionPrice.available,
+          optionLessonsValue: formData.get('optionLessons') === 'on',
+          comment: formData.get('comment')
+          // preferredVariant: 'string', // TODO: get preferred variant from form?
+        }
+      }))
+    }
+  }
+
+  disconnectedCallback () {
+    this.form.addEventListener('submit', this.submitListener)
```

Comment:
also remove the btn listener as well as creat the submit listener in the constructor and not connected callback.